### DISCUSSION TITLE
Creating roles and permission for optional team

### DISCUSSION THREAD

**Discussion by pritborteam** [spatie/laravel-permission#2752] [Q&A]
Docuemntation says we can have team_id as null if we want to create for a user not associated with any team. But it does not work when you have $team variable set to true in permissions.php file. Migrataion file does not have team_foreign_key nullable or migration file just checks based of $team variable true and false. I have a use case where not all users will have team. What to do?

**Comment by parallels999** [comment]
- https://github.com/spatie/laravel-permission/discussions/2738
